src/main/scala/com/fulcrumgenomics/vcf/MakeMixtureVcf.scala

```diff
@@ -0,0 +1,263 @@
+import htsjdk.variant.vcf._
+
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import scala.collection.mutable
+
+
+object MakeMixtureVcf {
+  /** The format field for the allele fraction to be written to. */
+  val AlleleFractionField = "AF"
+  /** The filter that is placed on records where the full genotype cannot be computed. */
+  val UnknownGtFilter     = "unknown_gt"
+
+  val HeaderLines = Seq[VCFHeaderLine](
+    new VCFFilterHeaderLine(UnknownGtFilter, "GT and AF not known due to one or more input samples being no-called."),
+    new VCFFormatHeaderLine("AF", VCFHeaderLineCount.A, VCFHeaderLineType.Float, "Alt allele fraction.")
+  )
+
+  /**
+    * Gets the genotype of the sample, substituting in a HomRef genotype for no-calls if required.
+    */
+  def genotypeOf(sample: String, ctx: VariantContext, noCallIsHomRef: Boolean): Genotype = {
+    val actual = ctx.getGenotype(sample)
+    if (actual.isNoCall && noCallIsHomRef) new GenotypeBuilder(sample, util.Arrays.asList(ctx.getReference)).make()
+    else actual
+  }
+
+  /** Makes a no-call genotype object with the provided sample name. */
+  def noCall(sampleName: String): Genotype = {
+    new GenotypeBuilder(sampleName).alleles(util.Arrays.asList(Allele.NO_CALL, Allele.NO_CALL)).make()
+  }
+
+  /** Builds a header that can be used to write a mixture VCF. */
+  def makeWriter(output: PathToVcf, in: VCFHeader, lines: Seq[VCFHeaderLine], samples: String*): VariantContextWriter = {
+    val header = new VCFHeader(Collections.emptySet[VCFHeaderLine](), util.Arrays.asList(samples:_*))
+
+    in.getFilterLines.foreach(header.addMetaDataLine)
+    in.getFormatHeaderLines.foreach(header.addMetaDataLine)
+    in.getInfoHeaderLines.foreach(header.addMetaDataLine)
+    lines.foreach(header.addMetaDataLine)
+
+    header.setSequenceDictionary(in.getSequenceDictionary)
+
+    val writer = new VariantContextWriterBuilder()
+      .setOutputFile(output.toFile)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .setReferenceDictionary(in.getSequenceDictionary)
+      .setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER)
+      .build()
+
+    writer.writeHeader(header)
+    writer
+  }
+}
+
+
+@clp(group=ClpGroups.VcfOrBcf, description=
+  """Creates a VCF with one sample whose genotypes are a mixture of other samples'.
+    |
+    |The input VCF must contain all samples to be mixed, and may optionally contain other samples.
+    |Sample mixtures can be specified in one of several ways:
+    |  1. --samples s1 s2 s3: specifies that the samples with names s1, s2 and s3 should be mixed equally.
+    |  2. --samples s1@0.1 s2@0.1 s3@0.8: specifies that the three samples should be mixed at 0.1, 0.1 and 0.8
+    |  3. --samples s1@0.1 s2 s3: specifies that s1 should form 0.1 of the mixture and that the remaining (0.9)
+    |                             should be split amongst s2 and s3
+    |  4. If no sample names are given, all samples in the input VCF will be used, mixing equally
```

Comment:
It's assumed that sample names are unique in the input VCF.  I looked at the VCF spec briefly and couldn't see if this was a requirement!